tmt/utils/__init__.py

```diff
@@ -5552,14 +5552,34 @@ def normalize_data_amount(
     logger: tmt.log.Logger,
 ) -> 'Size':
     from pint import Quantity
+    from pint.errors import DimensionalityError, UndefinedUnitError
 
     if isinstance(raw_value, Quantity):
-        return raw_value
+        # Validate existing quantity can be converted to bytes
+        try:
+            raw_value.to('bytes')
+            return raw_value
+        except DimensionalityError as exc:
+            raise NormalizationError(
+                key_address, raw_value, 'a valid data quantity (e.g., 1MB, 32MiB, 100KiB)'
+            ) from exc
 
     if isinstance(raw_value, str):
         import tmt.hardware
 
-        return tmt.hardware.UNITS(raw_value)
+        try:
+            quantity = tmt.hardware.UNITS(raw_value)
+            # Check unit compatibility by converting to bytes
+            quantity.to('bytes')
+            return quantity
+        except UndefinedUnitError as exc:
```

Comment:
```suggestion
        except (UndefinedUnitError, DimensionalityError) as exc:
```
But do we really care what error it passed or should it just catch any `PintError`?

Reply:
addressed: https://github.com/teemtee/tmt/commit/58ba408156027ec789c3201deb615c817d25b18a